File: Javascript/mergeSortedArray.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
    var r1 = m - 1;
    var r2 = n - 1;
    for (var i = m + n - 1; i >= 0; i--) {
        if (r1 >= 0 && r2 >= 0) {
            nums1[i] = nums1[r1] > nums2[r2] ? nums1[r1--] : nums2[r2--];
        } else if (r1 >= 0) {
            nums1[i] = nums1[r1--];
        } else {
            nums1[i] = nums2[r2--];
        }
    }
    return;
};

File: Java/mergeSortedArray.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int fin[] = new int[m+n];
        int j = 0;
        int k = 0;
        for(int i = 0; i < n+m; i++){
            if(j < m){
                if(k < n){
                    if(nums1[j] < nums2[k]){
                        fin[i] = nums1[j];
                        j++;
                    }else if(nums1[j] == nums2[k]){
                        fin[i] = nums1[j];
                        j++;
                    }else{
                        fin[i] = nums2[k];
                        k++;
                    }
                }else{
                    fin[i] = nums1[j];
                    j++;
                    }
            }else{
                if(k < n){
                    fin[i] = nums2[k];
                    k++;
                }
            }
        }
         for(int i = 0; i < m+n; i++){
             nums1[i] = fin[i];
         }
    }
}